src/workerd/jsg/observer.h

```diff
@@ -16,6 +17,58 @@ namespace v8 {
 
 namespace workerd::jsg {
 
+struct ResolveObserver {
+  virtual ~ResolveObserver() noexcept(false) { }
+
+  enum class Context {
+    BUNDLE,
+    BUILTIN,
+    BUILTIN_ONLY,
+  };
+
+  enum class Source {
+    // The resolve originated from a static import statement.
+    STATIC_IMPORT,
+    // The resolve originated from a dynamic import statement.
+    DYNAMIC_IMPORT,
+    // The resolve originated from a CommonJS require() call.
+    REQUIRE,
+    // The resolve originated from an internal direct call to
+    // the ModuleRegistry.
+    INTERNAL,
+    // The resolve originated from some other source (to be defined).
+    OTHER,
+  };
```

Comment:
Review note: The source is generally informational. It is included largely to support the fallback service use case where they need to know how exactly a module is being resolved.